### PR TITLE
libdvbcsa: add package

### PR DESCRIPTION
**Pull request by melias122**
###### Description of changes

Added libdvbcsa library. Needed for upcomming tvheadend udpate.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libdvbcsa/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "libdvbcsa";
+  version = "1.1.0";
+
+  src = fetchurl {
+    url = "http://get.videolan.org/libdvbcsa/${version}/${pname}-${version}.tar.gz";
+    sha256 = "4db78af5cdb2641dfb1136fe3531960a477c9e3e3b6ba19a2754d046af3f456d";
+  };
+
```
Comment (mweinelt) [inline pkgs/development/libraries/libdvbcsa/default.nix:11]:
```suggestion
  doCheck = true;
```

**Comment by risicle** [comment]
Commit name should be `libdvbcsa: init at 1.1.0`

**Comment by risicle** [comment]
Successfully built (with `doCheck = true` on macos 10.15 & nixos aarch64.

File: pkgs/development/libraries/libdvbcsa/default.nix (hunk 1 above)
Comment (melias122) [reply pkgs/development/libraries/libdvbcsa/default.nix:11]: Done

**Comment by mweinelt** [comment]
Following CONTRIBUTION.md we still need `meta.maintainers` set to a maintainers object.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -14348,4 +14348,10 @@
     github = "kuwii";
     githubId = 10705175;
   };
+  melias122 = {
+    name = "Martin Elias";
+    email = "martin+nixpkgs@elias.sx";
+    github = "melias122";
+    githubId = 1027766;
+  };
```
Comment (mweinelt) [inline maintainers/maintainer-list.nix:14356]: This needs to be a separate commit `maintainers: add melias122`.

File: pkgs/development/libraries/libdvbcsa/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  doCheck = true;
+
+  meta = {
+    description = "A free implementation of the DVB Common Scrambling Algorithm with encryption and decryption capabilities";
+    homepage = "http://www.videolan.org/developers/libdvbcsa.html";
+    platforms = lib.platforms.unix;
+    license = lib.licenses.gpl1;
+    maintainers = [ melias122 ];
```
Comment (mweinelt) [inline pkgs/development/libraries/libdvbcsa/default.nix:19]:
```suggestion
    maintainers = with lib.maintainers; [ melias122 ];
```

The maintainer objects are below `lib.maintainers`.

Comment (melias122) [reply pkgs/development/libraries/libdvbcsa/default.nix]: Done

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (melias122) [reply maintainers/maintainer-list.nix:14356]: Done
